Content.Shared/Jukebox/SharedJukeboxComponent.cs

```diff
@@ -0,0 +1,128 @@
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Jukebox;
+
+[NetworkedComponent, RegisterComponent]
+[Access(typeof(SharedJukeboxSystem))]
+public partial class JukeboxComponent : Component
+{
+    [DataField("playing")]
+    public bool Playing { get; set; }
+
+    [DataField("selectedSongId")]
+    public int SelectedSongID { get; set; }
+
+    [ViewVariables]
+    public MusicListDefinition? SelectedSong { get; set; }
+
+    [DataField("songTime")]
+    public float SongTime { get; set; }
+
+    [DataField("songStartTime")]
+    public float SongStartTime { get; set; }
+
+    [ViewVariables]
+    public IPlayingAudioStream? AudioStream { get; set; }
+
+    [ValidatePrototypeId<MusicListPrototype>]
+    public string MusicCollection = "MusicStandard";
+
+    [ViewVariables]
+    public MusicListPrototype JukeboxMusicCollection = default!;
+
+    [DataField("onState")]
+    public string? OnState;
+
+    [DataField("offState")]
+    public string? OffState;
+
+    [DataField("selectState")]
+    public string? SelectState;
```

Comment:
now datafields generate the name if not given as arguments, so use the short-form (apply this to the other fields). Also, storing these in the component its usually not necessary, cant you map the layers to an enum or something?